prov/gni/src/gnix_cm_nic.c

```diff
@@ -155,6 +158,12 @@ static int __process_datagram(struct gnix_datagram *dgram,
 				fi_strerror(-ret));
 			goto err;
 		}
+
+		ret = _gnix_cm_nic_progress(cm_nic);
```

Comment:
All calls to cm_nic_progress() were moved from points in gnix_vc.c to this file.